classes/plugin.php

```diff
@@ -78,60 +79,148 @@ public function add_feed_namespace() {
 		}
 
-		$options = get_option( $this->plugin_slug . '_options', array(
-			'author'      => true,
-			'description' => true,
-		) );
-
-		$thumbnail = get_post( get_post_thumbnail_id( $post->ID ) );
-		$thumbnail = apply_filters( 'wp_feed_post_thumbnail_image', $thumbnail );
+		/**
+		 * List of attachment media images.
+		 *
+		 * @since 2.1.2
+		 *
+		 * @param \WP_Post[] $images An array of attachment post objects
+		 */
+		$images = apply_filters( 'wp_feed_post_thumbnail_images', $images );
 
-		if ( ! $thumbnail instanceof WP_Post ) {
+		if ( empty( $images ) ) {
 			return;
 		}
 
-		$img_attr       = wp_get_attachment_image_src( $thumbnail->ID, apply_filters( 'wp_feed_post_thumbnail_image_size_full', 'full' ) );
-		$img_attr_thumb = wp_get_attachment_image_src( $thumbnail->ID, apply_filters( 'wp_feed_post_thumbnail_image_size_thumbnail', 'thumbnail' ) );
-		?>
-		<media:content url="<?php echo esc_url( $img_attr[0] ); ?>" type="<?php echo esc_attr( $thumbnail->post_mime_type ); ?>" medium="image" width="<?php echo absint( $img_attr[1] ); ?>" height="<?php echo absint( $img_attr[2] ); ?>">
-			<media:title type="plain">
-				<![CDATA[<?php echo sanitize_text_field( apply_filters( 'wp_feed_post_thumbnail_title', $thumbnail->post_title ) ); ?>]]>
-			</media:title>
-			<media:thumbnail url="<?php echo esc_url( $img_attr_thumb[0] ); ?>" width="<?php echo absint( $img_attr_thumb[1] ); ?>" height="<?php echo absint( $img_attr_thumb[2] ); ?>" />
-			<?php if ( isset( $options['description'] ) && $options['description'] ) : ?>
-				<media:description type="plain">
-					<![CDATA[<?php echo wp_kses_post( apply_filters( 'wp_feed_post_thumbnail_description', $thumbnail->post_content ) ); ?>]]>
-				</media:description>
-			<?php endif; ?>
-			<?php if ( isset( $options['author'] ) && $options['author'] ) : ?>
-				<media:copyright>
-					<?php echo esc_html( apply_filters( 'wp_feed_post_thumbnail_author', get_the_author() ) ); ?>
-				</media:copyright>
-			<?php endif; ?>
-		</media:content>
-		<?php
+		$options = get_option( $this->plugin_slug . '_options' );
+
+		foreach ( $images as $image ) :
+			if ( ! $image instanceof WP_Post ) {
+				return;
+			}
+
+			/**
+			 * Image size for the primary image.
+			 *
+			 * @since 1.0.0
+			 * @since 2.1.2 The `$image` parameter was added.
+			 *
+			 * @param string $size Image size. Default: full.
+			 * @param \WP_Post $image Attachment post object.
+			 */
+			$img_attr = wp_get_attachment_image_src( $image->ID, apply_filters( 'wp_feed_post_thumbnail_image_size_full', 'full', $image ) );
+
+			/**
+			 * Image size for the thumbnail.
+			 *
+			 * @since 1.0.0
+			 * @since 2.1.2 The `$image` parameter was added.
+			 *
+			 * @param string $size Thumbnail image size. Default: thumbnail.
+			 * @param \WP_Post $image Attachment post object.
+			 */
+			$img_attr_thumb = wp_get_attachment_image_src( $image->ID, apply_filters( 'wp_feed_post_thumbnail_image_size_thumbnail', 'thumbnail', $image ) );
```

Comment:
Better to store the result for the filter in a variable first so the DocBlock is right next to the filter.